docs: Reorganize 12-factor how-to guides (#877)
canonical/rockcraft: docs/how-to/manage-web-app-rock/index.rst

Reviewer: It looks like we're both converging toward the same style for indexes. :)

I'd like if we could be consistent in the code patterns for these. Here's what I did in Snapcraft: https://github.com/canonical/snapcraft/blob/main/docs/how-to/index.rst?plain=1

How to manage a 12-Factor app rock
**********************************

These guides walk you through all the ways you can manage 12-factor app rocks,
from initialization to usage.

Initialization
--------------

> Use the pre-defined Rockcraft extension to cater the initial rock files to
> your specific web app framework.

Reviewer: ```suggestion
To start, use one of the provided web framework extensions to scaffold the initial
Rockcraft project file.
```

If "scaffold" feels like a stretch, "pre-populate" also works.

.. toctree::

>     :maxdepth: 2

Author: I'll end up removing this when I adopt the same style of formatting as https://github.com/canonical/snapcraft/blob/main/docs/how-to/index.rst?plain=1, but now I'm wondering if you meant something different when you said "code pattern"...

>     :maxdepth: 2

Reviewer: Nope, you're correct!

    Initialize a 12-factor app rock <init-web-app-rock>

Configuration
-------------

> Once your 12-factor app rock has been set up, you can customize many aspects

Reviewer: ```suggestion
Once your 12-factor app rock is initialized, you can customize many aspects
```

> Once your 12-factor app rock has been set up, you can customize many aspects

Reviewer: I'm wary of "customize" – it could have unintended allusions. Could we instead say "arrange its contents"?
